### DISCUSSION TITLE
Viewing HTML files inline instead of downloading

### DISCUSSION THREAD

**Discussion by marrojwala** [filebrowser/filebrowser#988] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have hosted a bunch of HTML files on the filebrowser ( which provides brilliant organizing) and when I open them, they get downloaded instead of being viewed.

**Describe the solution you'd like**
I would like them to be opened inline when double-clicked instead of downloading them.

**Describe alternatives you've considered**
I used jupyter server to host these files which when I open there, render the html files on the browser itself, but jupyter server lacks the ease of usage and access management that filebrowser provides.

**Comment by o1egl** [comment]
Which version of filebrowser version/os/browser you use? Can't reproduce this issue on my local machine.

**Comment by marrojwala** [comment]
File Browser 2.1.2, running on Centos 7, using caddy's proxy directive to access it on my local system (macOS 10.15.5). I have tested it on both Chrome and Firefox.

**Comment by o1egl** [comment]
I think caddy cuts some headers. Can you try to access filebrowser directly?

**Comment by marrojwala** [comment]
I tried locally on macOS 10.15.5, when I open the file (28MB), it still prompts me to download. I tried it with a smaller file (210KB), it opens the HTML code of the file as shown below
![Screen Shot 2020-06-17 at 4 25 20 PM](https://user-images.githubusercontent.com/50930961/84960735-4452c400-b0b7-11ea-8cbc-27777fe5ab2f.png)

Ideally, I am hoping for this file to open in a rendered format as below
![Screen Shot 2020-06-17 at 4 25 42 PM](https://user-images.githubusercontent.com/50930961/84960738-487ee180-b0b7-11ea-8199-0ba8c5b39355.png)

**Comment by o1egl** [comment]
I don't think rendering an HTML document on click is the expected behavior. However we can think about adding an additional preview mode for html documents.

**Comment by marrojwala** [comment]
I can tinker around if you think the implementation is simple enough. If you could point me to the right code location, that would be great. I have a basic knowledge of both Go and Vue.js

**Comment by copart** [comment]
Would be great if HTML documents had a "preview" mode like markdown (MD)... however, still the default should be to edit the HTML.
